### PR TITLE
chore(vertex_ai): ensure generated vertex version is auto updated when versioning 

### PR DESCRIPTION
**Pull request by russellwheatley**
## Description

generate version using melos `preCommit` hook. 

I also ran the script manually which correctly [updated the version](https://github.com/firebase/flutterfire/pull/12904/files#diff-c8558e899c587dfa1f90da2d351c6b5b667a85f0ac6957b13c1033063a8d2feeR16).

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database](https://github.com/flutter/flutter/issues). Indicate, which of these issues are resolved or fixed by this PR. Note that you'll have to prefix the issue numbers with flutter/flutter#.*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [ ] I read and followed the [Flutter Style Guide].
- [ ] I signed the [CLA].
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/firebase/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Review by TarekkMA — APPROVED** [review]

File: melos.yaml [hunk 1]
```diff
@@ -15,6 +15,10 @@ command:
     # branch: master
     # Additionally build a changelog at the root of the workspace.
     workspaceChangelog: true
+    hooks:
+      preCommit: |
+        dart run scripts/generate_version.dart && \
```
Comment (Lyokone) [inline melos.yaml:20]: Should it be generate_vertexai_version ?

**Review by Salakar — APPROVED** [review]

File: melos.yaml (hunk 1 above)
Comment (russellwheatley) [reply melos.yaml:20]: good catch, I think I went too wild with `cmd + z` 👍

Comment (Salakar) [reply melos.yaml]: "vertexai" is the correct naming if we're being pedantic 😆

File: melos.yaml [hunk 2]
```diff
@@ -15,6 +15,10 @@ command:
     # branch: master
     # Additionally build a changelog at the root of the workspace.
     workspaceChangelog: true
+    hooks:
+      preCommit: |
+        dart run scripts/generate_vertex_version.dart && \
```
Comment (Salakar) [inline melos.yaml:20]:
```suggestion
        dart run scripts/generate_vertexai_version.dart && \
```

**Review by Lyokone — APPROVED** [review]

**Pull request merged**